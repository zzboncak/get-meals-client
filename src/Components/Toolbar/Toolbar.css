.toolbar {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background: rgb(248, 219, 180);
  height: 64px;

}

.toolbar_navigation {
  display: flex;
  height: 100%;
  align-items: center;
  padding: 0 1rem;
}
.toolbar_logo {
  margin-left: 1rem;
}

.toolbar_logo a {
color: black;
text-decoration: none;
font-size: 1.5rem;
}

.spacer {
  flex: 1;
}

.toolbar_navigation-items ul {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
}

.toolbar_navigation-items li {
  padding: 0 0.5rem;
}


.toolbar_navigation-items a {
  color: black;
  text-decoration: none;
}

.toolbar_navigation-items a:hover, 
.toolbar_navigation-items a:active {

  color: #fa923f;

}

.toolbar_toggle-button {
  height: 75%;
}

@media (max-width: 768px) {
  .toolbar_navigation-items {
    display: none;
  }


}

@media (min-width: 769px) {
  .toolbar_toggle-button {
    display: none;
  }

  .toolbar__logo {
    margin-left: 0;
  }
}

img {
  width: 80px;
  height: 70px;
}