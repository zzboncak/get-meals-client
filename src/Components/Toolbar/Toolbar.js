import React from 'react';
import DrawerToggleButton from '../SideDrawer/DrawerToggleButton';
import { NavLink } from "react-router-dom";
import './Toolbar.css';


const toolbar = props => (
  <header className="toolbar">
  <nav className="toolbar_navigation">
  <div className="toolbar_toggle-button">
  <DrawerToggleButton click={props.drawerClickHandler} />
  </div>
  <div className="toolbar_logo"><a href="/">GET MEALS</a></div>
  <div className="spacer" />
  <div className="toolbar_navigation-items">
  <ul>
  
  <li><NavLink to="/">Home</NavLink></li>
  <li><NavLink to="/add-location">Add Location</NavLink></li>
  <li><NavLink to="/About">About Us</NavLink></li>
  </ul></div>
  </nav>
    </header>
);

export default toolbar;