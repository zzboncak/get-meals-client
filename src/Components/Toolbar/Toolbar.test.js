import React from 'react';
import ReactDOM from 'react-dom';
import Toolbar from './Toolbar';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe(`Toolbar component`, () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter>
        <Toolbar />
    </BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <Toolbar />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
