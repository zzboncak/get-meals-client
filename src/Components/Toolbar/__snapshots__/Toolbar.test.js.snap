// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component renders the component as expected 1`] = `
<header
  className="toolbar"
>
  <nav
    className="toolbar_navigation"
  >
    <div
      className="toolbar_toggle-button"
    >
      <button
        className="toggle-button"
      >
        <div
          className="toggle-button__line"
        />
        <div
          className="toggle-button__line"
        />
        <div
          className="toggle-button__line"
        />
      </button>
    </div>
    <div
      className="toolbar_logo"
    >
      <a
        href="/"
      >
        GET MEALS
      </a>
    </div>
    <div
      className="spacer"
    />
    <div
      className="toolbar_navigation-items"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/add-location"
            onClick={[Function]}
          >
            Add Location
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/About"
            onClick={[Function]}
          >
            About Us
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
